lib/server.js

```diff
@@ -675,12 +675,18 @@ class WebServer {
             //  alarm_arm_away, alarm_arm_home, alarm_arm_night, alarm_arm_custom_bypass, alarm_disarm (code will be sent)
             this.log.debug(data.service + ': ' + id + ' = ' + data.service_data.code ? 'XXX' : 'none');
             this.adapter.getForeignObject(id, (err, obj) => {
-                if (obj.native.alarm_code && obj.native.alarm_code.toString() !== data.service_data.code.toString()) {
+                this.log.debug("native" + JSON.stringify(obj.native));
+                if (data.service_data == null || data.service_data.code == null) {
+                    this._sendResponse(ws, data.id, {result: false, error: 'code is empty'});
+                } else if (!obj.native.alarm_code || !data.service_data.code) {
+                    this.log.warn(`No code is defined! To provide the code add to object ${id} native.alarm_code with desired code`);
+                } else if (obj.native.alarm_code.toString() !== data.service_data.code.toString()) {
                     this._sendResponse(ws, data.id, {result: false, error: 'invalid code'});
                 } else {
-                    this.log.warn(`No code is defined! To provide the code add to object ${id} native.alarm_code with desired code`);
-                    this.adapter.setForeignState(id, data.service.startsWith('alarm_arm'), false, {user}, () =>
-                        this._sendResponse(ws, data.id));
+                    this.adapter.setForeignState(id + '.arm_state', data.service, false, {user}, () => {
```

Comment:
Ok, now I see why you did set the Id the way you did... 
We should not force the id-relation of multiple foreign states in a specific way, especially not in a way that is not officially supported.

An improvised fix could be to just use our state here for the full arm state (i.e. use `this.adapter.namespace + '.control.alarm_arm_state'` as id here).

Even better would be to improve processManualEntity for alarm_control_panel so that it checks the type, `boolean` should only set true/false, `string` should set full arm_state. Also common.states should be supported. Or we can allow to set some (until we improve UI hidden) things in the custom settings, too. Like a second state for the full alarm_text or something.